plugins/image-prioritizer/class-image-prioritizer-img-tag-visitor.php

```diff
@@ -90,6 +104,23 @@ public function __invoke( OD_Tag_Visitor_Context $context ): bool {
 		}
 		// TODO: If an image is visible in one breakpoint but not another, add loading=lazy AND add a regular-priority preload link with media queries (unless LCP in which case it should already have a fetchpriority=high link) so that the image won't be eagerly-loaded for viewports on which it is not shown.
 
+		// Ensure that sizes=auto is set properly.
+		$sizes = $processor->get_attribute( 'sizes' );
+		if ( is_string( $sizes ) ) {
+			$is_lazy  = 'lazy' === $get_attribute_value( 'loading' );
+			$has_auto = $this->sizes_attribute_includes_valid_auto( $sizes );
```

Comment:
I think having yet another implementation of this method here is not great and overkill. Since the primary purpose of this is to support the feature in Core, I think we should look for the Core function (see https://github.com/WordPress/wordpress-develop/pull/7253) and use it if present. Otherwise, let's just go with something simple inline.
```suggestion
			if ( function_exists( 'wp_sizes_attribute_includes_valid_auto' ) ) {
				$has_auto = wp_sizes_attribute_includes_valid_auto( $sizes );
			} else {
				$has_auto = 'auto' === $sizes || str_starts_with( $sizes, 'auto,' );
			}
```

Of course the fallback is not 100% reliable, but it's also only a fallback, and it'll work perfectly well to cater for what the "Enhanced Responsive Images" plugin does. We're mostly doing this to support Core and that plugin, so I think this is a better solution in scope of this plugin.

Related: https://github.com/WordPress/performance/pull/1476/files#r1747386984

Reply:
How about 775ab186dad9e6854d15cfc4d6f0aed295abf18c?

Reply:
LGTM 👍